packages/opentelemetry-api/src/trace/spancontext-utils.ts

```diff
@@ -13,24 +13,34 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { SpanContext } from './span_context';
+import { TraceFlags } from './trace_flags';
 
-import { SpanContext, TraceFlags } from '@opentelemetry/api';
-
-export const INVALID_SPANID = '0';
-export const INVALID_TRACEID = '0';
+const VALID_TRACEID_REGEX = /^([0-9a-f]{16}){2}$/i;
+const VALID_SPANID_REGEX = /^[0-9a-f]{16}$/i;
+const INVALID_ID_REGEX = /^0+$/i;
+export const INVALID_SPANID = '0000000000000000';
+export const INVALID_TRACEID = '00000000000000000000000000000000';
 export const INVALID_SPAN_CONTEXT: SpanContext = {
   traceId: INVALID_TRACEID,
   spanId: INVALID_SPANID,
   traceFlags: TraceFlags.NONE,
 };
 
+export function isValidTraceId(traceId: string): boolean {
+  return VALID_TRACEID_REGEX.test(traceId) && !INVALID_ID_REGEX.test(traceId);
+}
+
+export function isValidSpanId(spanId: string): boolean {
+  return VALID_SPANID_REGEX.test(spanId) && !INVALID_ID_REGEX.test(spanId);
+}
+
 /**
  * Returns true if this {@link SpanContext} is valid.
  * @return true if this {@link SpanContext} is valid.
  */
 export function isValid(spanContext: SpanContext): boolean {
   return (
-    spanContext.traceId !== INVALID_TRACEID &&
-    spanContext.spanId !== INVALID_SPANID
+    isValidTraceId(spanContext.traceId) && isValidSpanId(spanContext.spanId)
```

Comment:
valid trace ids are 32 hex characters and span ids 16 hex characters, not sure why we weren't checking this before.

Reply:
We weren't checking before because we know IDs we generate are the correct length, and the propagator checks length on incoming ids separately. The only invalid id we can generate is the all-zero id.